src/duqtools/duqmap.py

```diff
@@ -0,0 +1,57 @@
+from inspect import signature
+from pathlib import Path
+from typing import Any, Callable, List, Optional
+
+from .ids import ImasHandle
+from .models import Locations, Run
+
+
+def _duqmap(function: Callable[[Any], Any],
+            convert: Callable[[Run], Any],
+            runs: Optional[List[Any]] = None) -> List[Any]:
+    if not runs:
+        runs = Locations().runs
+
+    ret = []
+    for run in runs:
+        if isinstance(run, Run):
+            pass
+        elif isinstance(run, Path):
+            run = Run.from_Path(run)
+        else:
+            raise NotImplementedError(
+                f'Dont know how to convert: {type(run)} {run}, to Run')
+        ret.append(function(convert(run)))
+    return ret
+
+
+def duqmap_run(function: Callable[[Run], Any], **kwargs) -> List[Any]:
+    return _duqmap(function, lambda x: x, **kwargs)
+
+
+def duqmap_imas(function: Callable[[ImasHandle], Any], **kwargs) -> List[Any]:
+    return _duqmap(function, Run.to_ImasHandle, **kwargs)
```

Comment:
This is confusing to me. `Run.to_ImasHandle` is not a static method, I guess it works, but it feels dirty to do it in this way.

How about this:

```suggestion
    def to_imas_handle(run):
        return run.to_imas_handle()
    return _duqmap(function, to_imas_handle, **kwargs)
```